refinery/core/models.py

```diff
@@ -393,11 +395,18 @@ def get_groups(self, changeonly=False, readonly=False):
             group["id"] = group["group"].id
             group["change"] = False
             group["read"] = False
+            if self._meta.verbose_name == 'dataset':
+                group["read_meta"] = False
+
             for permission in permission_list:
                 if permission.startswith("change"):
                     group["change"] = True
-                if permission.startswith("read"):
+                if permission.startswith("read") and not \
+                        permission.startswith("read_meta"):
                     group["read"] = True
+                if permission.startswith("read_meta"):
+                    group["read_meta"] = True
```

Comment:
Maybe reverse these to have one less check?
```
                if permission.startswith("read_meta_"):
                    group["read_meta"] = True
                elif permission.startswith("read_"):
                     group["read"] = True
```
(Small preference for including as much of the prefix as possible.)

Reply:
unsure why these weren't elif to begin with